service/start.go

```diff
@@ -89,22 +91,17 @@ func (svc *service) startNostr(ctx context.Context, encryptionKey string) error
 				}).WithError(err).Error("Failed to connect to relay")
 				continue
 			}
-
 			waitToReconnectSeconds = 0
 
-			//publish event with NIP-47 info
-			err = svc.nip47Service.PublishNip47Info(ctx, relay, svc.lnClient)
-			if err != nil {
-				logger.Logger.WithError(err).Error("Could not publish NIP47 info")
-			}
+			// start each app wallet subscription which have a child derived wallet key
+			svc.startAllExistingAppsWalletSubscriptions(ctx, relay)
 
-			logger.Logger.Info("Subscribing to events")
-			sub, err := relay.Subscribe(ctx, svc.createFilters(svc.keys.GetNostrPublicKey()))
-			if err != nil {
-				logger.Logger.WithError(err).Error("Failed to subscribe to events")
-				continue
-			}
-			err = svc.StartSubscription(sub.Context, sub)
+			// register a subscriber for events of "app_created" which handles creation of nostr subscription for new app
+			svc.eventPublisher.RegisterSubscriber(&createAppConsumer{svc: svc, relay: relay})
+
+			// legacy single wallet subscription - only subscribe once for all legacy apps
```

Comment:
change to only set this up if at least one legacy app exists

Reply:
This is consumer for `app_created` events, so one generally always is needed to handle app creation right?